components/QueueEdit.js

```diff
@@ -0,0 +1,178 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import {
+  Col,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Form,
+  FormGroup,
+  Input,
+  Label,
+  Button,
+  FormFeedback,
+} from 'reactstrap'
+
+const fields = [
+  {
+    name: 'name',
+    minLength: 1,
+  },
+  {
+    name: 'location',
+    minLength: 1,
+  },
+]
+
+const isValid = error => (error === undefined ? undefined : error === '')
+
+class QueueEdit extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      name: '',
+      location: '',
+      fixedLocation: false,
+      isFieldValid: {},
+    }
+
+    this.handleInputChange = this.handleInputChange.bind(this)
+    this.handleSubmitEdit = this.handleSubmitEdit.bind(this)
+    this.handleCancel = this.handleCancel.bind(this)
+    this.handleModalExit = this.handleModalExit.bind(this)
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.queue) {
+      this.setState({
+        name: nextProps.queue.name,
+        location: nextProps.queue.location,
+        fixedLocation: nextProps.queue.fixedLocation,
+        isFieldValid: {},
+      })
+    }
+  }
+
+  handleInputChange(event) {
+    this.setState({
+      [event.target.name]: event.target.value,
+    })
+  }
+
+  handleCancel() {
+    this.props.onCancel()
+  }
+
+  handleSubmitEdit() {
+    const isFieldValid = {}
+    let valid = true
+
+    fields.forEach(({ name, minLength }) => {
```

Comment:
For a more concise version of the check on line 84, I've been doing something like this in the loop:

```js
if (name === 'location' && !this.props.queue.fixedLocation) return
```

Reply:
No, not at all! I fully agree that code should be as concise as possible. Thank you for pointing that out! Please see the latest commit.